cairo example does not work

Hi, the cairo example for me does not draw anything, it crates the transparent window but doesn't draw anything
Console output
```
MapNotify(MapNotifyEvent { response_type: 19, sequence: 26, event: 96468992, window: 96468992, override_redirect: false }))
Got an unknown event
Expose(ExposeEvent { response_type: 12, sequence: 26, window: 96468992, x: 0, y: 0, width: 100, height: 100, count: 0 }))
ConfigureNotify(ConfigureNotifyEvent { response_type: 150, sequence: 26, event: 96468992, window: 96468992, above_sibling: 0, x: 910, y: 479, width: 100, height: 100, border_width: 0, override_redirect: false }))
UnmapNotify(UnmapNotifyEvent { response_type: 18, sequence: 52, event: 96468992, window: 96468992, from_configure: false }))
Got an unknown event
MapNotify(MapNotifyEvent { response_type: 19, sequence: 52, event: 96468992, window: 96468992, override_redirect: false }))
Got an unknown event
Expose(ExposeEvent { response_type: 12, sequence: 52, window: 96468992, x: 0, y: 0, width: 100, height: 100, count: 0 }))
```

> [#26 (comment)](https://github.com/psychon/x11rb/issues/26#issuecomment-1193134743) uses
> 
> ```rust
> use x11rb::protocol::shape::{self, ConnectionExt as _};
> ```
> 
> The `self`-part makes e.g. `shape::SO:SET` work. The `ConnectionExt` trait makes `conn.shape_rectangles()` work.

That's weird, cargo won't build it
```
error[E0432]: unresolved imports `x11rb::protocol::shape`, `x11rb::protocol::shape`
  --> cairo-example/src/main.rs:18:22
   |
18 | use x11rb::protocol::shape::{self, ConnectionExt as _};
   |                      ^^^^^   ^^^^ no `shape` in `protocol`
   |                      |
   |                      could not find `shape` in `protocol`
```

--- a/cairo-example/src/main.rs
+++ b/cairo-example/src/main.rs
@@ -242,10 +242,15 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     let atoms = AtomCollection::new(&conn)?.reply()?;
     let (mut width, mut height) = (100, 100);
     let (depth, visualid) = choose_visual(&conn, screen_num)?;
+    println!("Using visual {:#x} with depth {}", visualid, depth);
 
     // Check if a composite manager is running. In a real application, we should also react to a
     // composite manager starting/stopping at runtime.
     let transparency = composite_manager_running(&conn, screen_num)?;
+    println!(
+        "Composite manager running / working transparency: {:?}",
+        transparency
+    );
 
     let window = create_window(&conn, screen, &atoms, (width, height), depth, visualid)?;
 

--- a/cairo-example/src/main.rs
+++ b/cairo-example/src/main.rs
@@ -153,7 +153,6 @@ where
         visual_id,
         &win_aux,
     )?;
-    conn.free_colormap(colormap)?;
 
     let title = "Simple Window";
     conn.change_property8(

